opm/grid/cpgrid/CpGrid.cpp

```diff
@@ -2236,10 +2235,10 @@ CpGrid::defineChildToParentAndIdxInParentCell(std::map<std::array<int,2>,std::ar
     for (int shiftedLevel = 0; shiftedLevel < static_cast<int>(refined_cell_count_vec.size()); ++shiftedLevel) {
         refined_child_to_parent_cells_vec[shiftedLevel].resize(refined_cell_count_vec[shiftedLevel]);
         refined_cell_to_idxInParentCell_vec[shiftedLevel].resize(refined_cell_count_vec[shiftedLevel]);
-        const auto& level =  shiftedLevel + startingGridIdx +1;
+        const auto& level =  shiftedLevel + (this->maxLevel()) +1;
```

Comment:
This change seems unrelated. Please double check and explain.

Reply:
It is now correct, I would say. I should have added a comment though. When calling multiple times adapt, or refining a grid that has been already refined, "startingGridIdx" is the integer to access "the current view data" that will be deleted once you refine/adapt for a second/third/... time. 
In the begging, I was not aware of this deleting process, and I wrote the code (in one of my first interpretations) in such a way that we still store those mixed grids, which is not the case in master branch (and in my merged PRs). 

Maybe this should have been part of a tiny PR called bugfix or similar, this indeed it is not related to passing maps as const references.

Reply:
To illustrate, if you have a grid and refine it once adding 2 levels, then the first mixed grid is stored in data_ in the entry 3.
data_[0] = level 0, data_[1] = level 1 grid, data_[2] = level 2, data_[3] = "current view data"
Now, you mark elements to be refined, in "current view data". Later on, this "current view data" is deleted/replaced for the refined level grid 3. Then
data_[0] = level 0, data_[1] = level 1 grid, data_[2] = level 2, data_[3] = level 3, data_[4] = (new) "current view data".


Therefore, old line (in this example), with shiftedLEvel =0 e.g.:
level = startingGridIdx + 1 =  3 + 1 = 4 which is wrong,
new line:
level = (this->maxLevel()) + 1 = 2+ 1 =3 which is correct!

I hope this helps!

Reply:
Thanks. So this is a fix to a bug, that was not triggered by any of our tests.

Reply:
Exactly. Found out thanks to .at() which from now, I'll keep it in mind!